pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,534 @@
+			let user: T::AccountId = ensure_signed(origin)?;
+			//check if given id valid or not
+			ensure!(
+				<InitializeRewards<T>>::contains_key(reward_id),
+				Error::<T>::RewardIdNotRegister
+			);
+
+			//check if user is added in reward list
+			ensure!(<Distributor<T>>::contains_key(reward_id, &user), Error::<T>::UserNotEligible);
+
+			<Distributor<T>>::mutate(reward_id, user.clone(), |user_reward_info| {
+				if let Some(user_reward_info) = user_reward_info {
+					// only unlock reward if current block greater than or equal to the starting
+					// block of reward
+					if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+						ensure!(
+							reward_info.start_block.saturated_into::<u128>() <=
+								<frame_system::Pallet<T>>::block_number()
+									.saturated_into::<u128>(),
+							Error::<T>::RewardsCannotBeUnlockYet
+						);
+					}
+
+					//check if user already unlocked the rewards
+					ensure!(!user_reward_info.is_initialized, Error::<T>::RewardsAlreadyUnlocked);
+					//transfer funds from pallet account to users account
+					ensure!(
+						Self::transfer_pdex_rewards(
+							&Self::get_pallet_account(),
+							&user,
+							user_reward_info.total_reward_amount
+						)
+						.is_ok(),
+						Error::<T>::TransferFailed
+					);
+
+					//lock funds in users account
+					T::NativeCurrency::set_lock(
+						REWARDS_LOCK_ID,
+						&user,
+						user_reward_info.total_reward_amount.saturated_into(),
+						WithdrawReasons::TRANSFER,
+					);
+					user_reward_info.is_initialized = true;
+					Ok(())
+				} else {
+					//sanity check
+					Err(Error::<T>::UserNotEligible)
+				}
+			})?;
+			Self::deposit_event(Event::UserUnlockedReward { user, reward_id });
+			Ok(())
+		}
+
+		/// The user will use the extrinsic to claim rewards
+		/// # Parameters
+		/// * `origin`: The users address which has been mapped to reward id
+		/// * `id`: The reward id
+		#[pallet::weight(10_000)]
+		pub fn claim(origin: OriginFor<T>, reward_id: u32) -> DispatchResult {
+			let user: T::AccountId = ensure_signed(origin)?;
+
+			//check if given id valid or not
+			ensure!(
+				<InitializeRewards<T>>::contains_key(reward_id),
+				Error::<T>::RewardIdNotRegister
+			);
+
+			//check if user is added in reward list
+			ensure!(<Distributor<T>>::contains_key(reward_id, &user), Error::<T>::UserNotEligible);
+
+			<Distributor<T>>::mutate(reward_id, user.clone(), |user_reward_info| {
+				if let Some(reward_info) = <InitializeRewards<T>>::get(reward_id) {
+					if let Some(user_reward_info) = user_reward_info {
+						//check if user has initialize rewards or not
+						ensure!(
+							user_reward_info.is_initialized,
+							Error::<T>::UserHasNotInitializeClaimRewards
+						);
+
+						//ensure that all rewards are not already claimed
+						ensure!(
+							user_reward_info.claim_amount.saturated_into::<u128>() !=
+								user_reward_info.total_reward_amount.saturated_into::<u128>(),
```

Comment:
Bigger is also not equal. Set it to claimed minus current is bigger than curren. Unclaimed must be enough to cover current claim

Reply:
Fixed in the latest commit